Add database support and performance optimizations

wryuin opened the pull request:
Region data persistence through server restarts
Reduced WorldGuard region lookups
Memory usage optimizations
Better performance with large numbers of regions
Added DatabaseManager using HikariCP and SQLite for persistent storage
Created OptimizationManager for caching and performance improvements
Modified existing classes to work with the database
Added asynchronous operations for better performance

wryuin commented:
Region data persistence through server restarts
Reduced WorldGuard region lookups
Memory usage optimizations
Better performance with large numbers of regions
Added DatabaseManager using HikariCP and SQLite for persistent storage
Created OptimizationManager for caching and performance improvements
Modified existing classes to work with the database
Added asynchronous operations for better performance

jerzean commented:
this needs to be reviewed by @espidev , ive looked through the code an it looks ok but
my question is why db support? PS pulls its data from the worldguard plugin which has its own internal DB is there some other purpose im missing?

wryuin commented:
@jerzean I don't know why I added it, I probably just didn't know about DB in WorldGuard, but many others may not have known that WorldGuard has a DB, that's why I added it.

espidev commented:
Regions are fetched from WorldGuard which has its own database already, and likely does some of its own caching internally. This appears to be adding an in-memory region cache, is this redundant to what WorldGuard already does?

We also have a UUID cache, to reduce queries to Mojang for UUID -> player names. This PR appears to change that to become an on-disk sqlite cache?

I'm curious if you have any statistics or benchmarks showing an substantial improvement in performance or memory reduction? Otherwise this introduces some complexity and the possibility of having to deal with cache corruption.

wryuin commented:
hey @espidev, i need a answer for this question. Can I support version 1.16.5 of this plugin?

espidev commented:
> hey @espidev, i need a answer for this question. Can I support version 1.16.5 of this plugin?

We stopped supporting that version and moved to newer API versions for new features. You can use an older version of the plugin (v2.8.5)

espidev commented:
> @espidev yes I know about this. But I'm want to updating this

You could fork the plugin to backport features and fixes, but there are no plans to continue support for that version in master

wryuin commented:
@espidev okay, do u have branch with code for 1.16.5 plugin?

espidev commented:
No branch, but you can check out the v2.8.5 git tag on the repo

wryuin commented:
> No branch, but you can check out the v2.8.5 git tag on the repo

ok bro, ty
